src/cascadia/TerminalControl/TermControlAutomationPeer.cpp

```diff
@@ -28,6 +28,32 @@ namespace XamlAutomation
     using winrt::Windows::UI::Xaml::Automation::Provider::ITextRangeProvider;
 }
 
+static constexpr wchar_t UNICODE_NEWLINE{ L'\n' };
+
+static inline std::wstring Sanitize(std::wstring_view text)

```

Comment:
add some doc comments foo'. What is this sanitizing? (It's stripping all control chars except `\n`, but let the reader know that!

Reply:
This function should be either static or inline, but not both. static means "local to this .cpp file" and "inline" means "deduplicate functions with the same name". I'd use static.

Additionally you should avoid pessimizing your code.
Your code currently assumes that the input string has a high amount of control characters we need to optimize for. After all you don't reserve() memory for the sanitized string and push characters one by one into it, which indicates that it's optimized for the case of filtering out a maximum amount of control characters.

If you optimize your code for filtering out control characters sometimes you only need to flip the logic from "create a filtered duplicate" to "create a duplicate which is filtered", because most of the time you don't need to filter anything, turning this into a pure duplication. Here's the code:

```cpp
static std::wstring Sanitize(std::wstring sanitized) // <-- fast memory duplication
{
    // vvv filtering which usually won't touch the string at all vvv
    // (The "erase-remove idiom".)
    sanitized.erase(std::remove_if(sanitized.begin(), sanitized.end(), [](wchar_t c) {
        return c < L' ' && c != L'\n';
    }), sanitized.end());
    return sanitized;
}
```

Lastly the only caller of `Sanitize()` has a fresh, mutable copy of the string.
As such the signature should be `static std::wstring Sanitize(std::wstring& sanitized)`.

Reply:
> Lastly the only caller of `Sanitize()` has a fresh, mutable copy of the string.
As such the signature could be `static std::wstring Sanitize(std::wstring& sanitized)`.

Should I take it a step further and just do `static void Sanitize(std::wstring& sanitized)`?

Reply:
Either that, or you change `NotifyNewOutput` to accept a `std::wstring_view` instead.

Reply:
> @lhecker Should I take it a step further and just do static void Sanitize(std::wstring& sanitized)?

Oh wait... I just realized I forgot to make the return type a `void`. Good catch. 😅
I intended to write `static void Sanitize(std::wstring& sanitized)`. 🙂